src/memprof_js_library.js

```diff
@@ -0,0 +1,9 @@
+mergeInto(LibraryManager.library, {
+  emscripten_trace_record_allocation: function (address, size) {
+    memProf.mallocProxy(address, size);
```

Comment:
The name `mallocProxy` feels a bit odd to me, since it does not immediately tell what the role of the function is. Perhaps something like `onMemoryAllocated`, `onMalloc`, or `traceMalloc` or something else that would hint that the runtime is calling out to a function to log the allocation. Also, should document somewhere that a) user is supposed to implement the function, and guide how to do it b) the function is called to inform/log about a malloc that has just occurred and c) the user shall not call to any C function from inside this function that might do a malloc or a free, or otherwise that would risk an infinite loop. Also, I am not sure if we should introduce a new global object by the name `memProf`? Perhaps it would be better to call these hooks on the Module object, i.e. `Module['onMemoryAllocated']`?

So perhaps it would be better that this would do something like

     emscripten_trace_record_allocation: function(address, size) {
         if (Module['onMalloc']) Module['onMalloc'](address, size);
    }

and similar for `emscripten_trace_record_free`.